main.go

```diff
@@ -1,9 +1,69 @@
 package main
 
 import (
+	"fmt"
+	"net/http"
+
+	_ "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	log "github.com/sirupsen/logrus"
+	flag "github.com/spf13/pflag"
+	config "github.com/spf13/viper"
 )
 
+func loglevel(level string) {
+	switch level {
+	case "error":
+		log.SetLevel(log.ErrorLevel)
+	case "warn":
+		log.SetLevel(log.WarnLevel)
+	case "info":
+		log.SetLevel(log.InfoLevel)
+	case "debug":
+		log.SetLevel(log.DebugLevel)
+	default:
+		log.Warnln("Unrecognized minimum log level; using 'info' as default")
+	}
+}
+
+func init() {
+	config.SetConfigName("sap-host-exporter")
+	config.AddConfigPath("./")
+	config.AddConfigPath("$HOME/.config/")
+	config.AddConfigPath("/etc/")
+	config.AddConfigPath("/usr/etc/")
+
+	flag.String("port", "9664", "The port number to listen on for HTTP requests")
```

Comment:
do we have reserved this port upstreaM?

Reply:
Ahh no, you're right, it's the same from the other project! We shall reserve a new one and update this diff!